libsolidity/formal/SMTChecker.cpp

```diff
@@ -607,7 +615,8 @@ void SMTChecker::checkCondition(
 			message << " for:\n";
 			solAssert(values.size() == expressionNames.size(), "");
 			for (size_t i = 0; i < values.size(); ++i)
-				message << "  " << expressionNames.at(i) << " = " << values.at(i) << "\n";
+				if (expressionsToEvaluate.at(i).name != values.at(i))
+					message << "  " << expressionNames.at(i) << " = " << values.at(i) << "\n";
```

Comment:
This `if` had to be added so that state variables that are not touched by a function don't appear in a counterexample, since they don't have a model. In that case, the solver returns the name of the SMT variable itself as the model. Right now this happens only for Bool. The reason is that when we set a Bool var to unknown, no constraint is added to the solver, therefore if the function doesn't do anything with this state var, it's just not there. Since integers have constraints for the unknown case, integer state variables will have a model even if they're not used.

So, we should choose one out of the following 3 cases for how counterexamples are shown:
1) As it is in this PR: unused state integer vars get a model, Bools don't.
2) Unused Bool state vars should have a model too. This case is probably simple to create, we just have to add the trivial constraint `(or var (not var))` in the unknown case.
3) No unused state vars should have models. This is more complicated and would require changes in the class `VariableUsage` or something similar

Reply:
I think it's fine this way.